cardano_node_tests/tests/test_cli.py

```diff
@@ -391,21 +391,70 @@ def test_ledger_state(self, cluster: clusterlib.ClusterLib):
 
     @allure.link(helpers.get_vcs_link())
     @pytest.mark.testnets
-    def test_stake_snapshot(self, cluster: clusterlib.ClusterLib, pool_ids: List[str]):
+    @pytest.mark.parametrize("option", ("single_pool", "multiple_pool", "all_pools", "total_stake"))
+    def test_stake_snapshot(
+        self,
+        cluster: clusterlib.ClusterLib,
+        option: str,
+        pool_ids: List[str],
+    ):
         """Test `query stake-snapshot`."""
         try:
-            stake_snapshot = cluster.g_query.get_stake_snapshot(stake_pool_id=pool_ids[0])
+            if option == "single_pool":
+                expected_pool_ids = [pool_ids[0]]
+                stake_snapshot = cluster.g_query.get_stake_snapshot(
+                    stake_pool_ids=expected_pool_ids
+                )
+            elif option == "multiple_pool":
+                expected_pool_ids = [pool_ids[0], pool_ids[1]]
+                stake_snapshot = cluster.g_query.get_stake_snapshot(
+                    stake_pool_ids=expected_pool_ids
+                )
+            elif option == "all_pools":
+                expected_pool_ids = pool_ids
+                stake_snapshot = cluster.g_query.get_stake_snapshot(all_stake_pools=True)
+            else:
+                expected_pool_ids = []
+                stake_snapshot = cluster.g_query.get_stake_snapshot()
         except json.decoder.JSONDecodeError as err:
             pytest.xfail(f"expected JSON, got CBOR - see node issue #3859: {err}")
 
-        assert {
-            "activeStakeGo",
-            "activeStakeMark",
-            "activeStakeSet",
-            "poolStakeGo",
-            "poolStakeMark",
-            "poolStakeSet",
-        }.issubset(stake_snapshot)
+        if clusterlib_utils.cli_has("query stake-snapshot --all-stake-pools"):
```

Comment:
It is better to look at the structure of returned dict instead of using `cli_has`.